src/components/molecules/NavSearchBar/NavSearchBar.scss

```diff
@@ -1,8 +1,19 @@
 @import "scss/constants.scss";
 
+$search-results-width: 18.125rem;
+
 .nav-search-links {
-  display: flex;
   position: relative;
+  width: $search-results-width;
+
+  .nav-search-input {
+    z-index: 2;
+  }
```

Comment:
Do we need to set a `z-index` here? (often we don't actually need them and just end up overusing them) If so, we should be using our `z()` helper (defined in `constants`)

Reply:
I used `z-index` to have input-div above the search-results. Search results are going from the top of the screen and if they both are on the same `z-index` level search-results overlaps input-div for a short period when appearing and hiding.

okay, I'll use the `z()`, thanks. :)

Reply:
fixed: https://github.com/sparkletown/sparkle/pull/1328/commits/e21e67445665543be0fab0f660743356141c30c7

Reply:
> if they both are on the same z-index level search-results overlaps input-div for a short period when appearing and hiding

Why does that happen? Is it because one (or both) are positioned outside of the 'normal flow' of things, or?

Reply:
if we don't add z-index to input-div, input-div and search-div will be on the same z-level. 
the html-structure is the following:

```html
<div className="NavSearchBar">
      <InputField className="NavSearchBar__search-input" ... />
      <div className="NavSearchBar__nav-dropdown">...</div>
</div>
```

let's say we don't apply `z-index`. The dropdown is mentioned later, so the second div will appear in front of the first div (if they're in the same place on the page).
Restyled dropdown comes from the top of the screen when you start typing. So during that transition it'll hide the input-div for a moment (same story when it hides). 
But if we apply higher z-index to input-div, it'll be in front of the dropdown-div no matter what. does that makes sense?

Reply:
Can we change the structure of them so that the dropdown comes first? And if so, does that fix it when we don't have any `z-index` set?

If we need to use the `z-index` for it then that's fine, and we can do that; but there are often better alternatives IMO (and we have traditionally tended to overuse `z-index`, which has made things harder to reason about/lead to weird bugs/etc)

Reply:
changed the order and removed unused z-index: https://github.com/sparkletown/sparkle/pull/1328/commits/16a2051fbc83a252406d90106ce1cc345aecf83a